test/integration/create-next-app/templates/matrix.test.ts

```diff
@@ -33,24 +33,30 @@ describe.each(['app', 'pages'] as const)(
       // eslint: ['--eslint', '--no-linter'],
       eslint: ['--eslint'],
 
-      srcDir: ['--src-dir', '--no-src-dir'],
+      // Trading test perf for robustness:
+      // srcDir and reactCompiler don't interact so we're testing them together
+      // instead of all permutations.
+      srcDirAndCompiler: [
+        '--src-dir --react-compiler',
+        '--no-src-dir --no-react-compiler',
+      ],
       tailwind: ['--tailwind', '--no-tailwind'],
 
       // shouldn't affect if the app builds or not
       // packageManager: ['--use-npm', '--use-pnpm', '--use-yarn', '--use-bun'],
     }
 
-    const getPermutations = <T>(items: T[][]): T[][] => {
+    const getCombinations = (items: string[][]): string[][] => {
```

Comment:
Unrelated change but I couldn't help myself. Permutations include different orders of the items in the sequence. We're just creating combinations here though. Testing each permutation would test the arg parser not create-next-app which would feel excessive to me.